src/modules/pages/components/RouteLayouts/UserNavigation.tsx

```diff
@@ -115,8 +145,177 @@ const UserNavigation = () => {
     }
   }, [userDataLoading, userLock, dispatch, ethereal]);
 
+  type NetworkHealthType = 'healthy' | 'poor' | 'critical';
+  const [networkHealth, setNetworkHealth] = useState<NetworkHealthType>(
+    'healthy',
+  );
+
+  const networkCheckInterval = 30 * 1000; // 15 seconds
```

Comment:
Interval should be higher, at least 1 to 2 minutes, having it every 30 seconds is very expensive.

You need to strike a balance between usefulness _(of the feature)_, and load on the servers.

Also, the comment is wrong